### PR TITLE
Research/v4.1.0 additional forecasters

### PR DESCRIPTION
**Pull request by Lars800**
V2 of the PR to introduce HybridForecaster, LGBMForecaster and LGBMLinearForecaster

These three new forecasters expand on existing OpenSTEF forecasters with new algorithms. 
Comments on initial PR were resolved, all forecaster specific logic is contained within the forecaster.py file. 

To enable multi quantile regression for SKlearn-like estimators, the MultiQuantileRegressor was introduced as a utility class. 
This provides a wrapper arround SKLearn with multi-quantile capabilities

### CODE REVIEW

File: packages/openstef-models/src/openstef_models/models/forecasting/lgbmlinear_forecaster.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+            "n_jobs": config.n_jobs,
+            **config.hyperparams.model_dump(),
+        }
+
+        self._lgbmlinear_model: MultiQuantileRegressor = MultiQuantileRegressor(
+            base_learner=LGBMRegressor,  # type: ignore
+            quantile_param="alpha",
+            hyperparams=lgbmlinear_params,
+            quantiles=[float(q) for q in config.quantiles],
+        )
```
Comment (egordm) [inline packages/openstef-models/src/openstef_models/models/forecasting/lgbmlinear_forecaster.py:252]: Not really a code related question. But LightGBM model doesn't require target variable normalization? We found that it makes a big difference in linear models.

Comment (Lars800) [reply packages/openstef-models/src/openstef_models/models/forecasting/lgbmlinear_forecaster.py:252]: Good remark! Perhaps it will help with the stability of the linear trees!
